compiler/aarch64/codegen/OMRTreeEvaluator.cpp

```diff
@@ -6400,10 +6400,74 @@ OMR::ARM64::TreeEvaluator::arraytranslateAndTestEvaluator(TR::Node *node, TR::Co
 
 TR::Register *
 OMR::ARM64::TreeEvaluator::arraytranslateEvaluator(TR::Node *node, TR::CodeGenerator *cg)
-	{
-	// TODO:ARM64: Enable TR::TreeEvaluator::arraytranslateEvaluator in compiler/aarch64/codegen/TreeEvaluatorTable.hpp when Implemented.
-	return OMR::ARM64::TreeEvaluator::unImpOpEvaluator(node, cg);
-	}
+   {
+   // tree looks as follows:
+   // arraytranslate
+   //    (0) input ptr
+   //    (1) output ptr
+   //    (2) translation table (dummy)
+   //    (3) stop character (terminal character, either 0xff00ff00 (ISO8859) or 0xff80ff80 (ASCII)
+   //    (4) input length (in elements)
+   //    (5) stopping char (dummy)
```

Comment:
Are the dummy nodes used by other architectures?  Just wondering why they're dummy on AArch64.

Reply:
It is different between p/x/aarch64 and z.  See the code below:
https://github.com/eclipse-omr/omr/blob/205c540402c01986f433a34412f9d7fcfe61d90f/compiler/optimizer/ValuePropagationCommon.cpp#L2535-L2565

The code for z from line 2543 generates a non-dummy node for child(2) at line 2562, while the non-z path generates a dummy `aconst 0` at line 2537.
AArch64 takes the non-z path.

Reply:
An example of the `arraytranslate` node for `TRTO255`:

```
 n819n    (  0)  treetop                                                                              [       0x11fa0afb0] bci=[0,13,180] rc=0 vc=2947 vn=- li=57 udi=- nc=1
 n818n    (  1)    arraytranslate  <arraytranslate>[#278  helper Method] [flags 0x400 0x0 ] (in GPR_0115) (char2byteXlate byteArrayXlate tableBackedByRawStorage )  [       0x11fa0af60] bci=[0,13,180] rc=1 vc=2947 vn=- li=57 udi=63488 nc=6 flg=0xa020
 n795n    (  1)      aladd (in GPR_0112) (X>=0 internalPtr )                                          [       0x11fa0a830] bci=[0,12,180] rc=1 vc=2947 vn=- li=57 udi=62432 nc=2 flg=0x8100
 n460n    (  0)        ==>aRegLoad (in &GPR_0016)
 n797n    (  0)        lsub (in GPR_0017) (X>=0 cannotOverflow )                                      [       0x11fa0a8d0] bci=[0,12,180] rc=0 vc=2947 vn=- li=57 udi=14464 nc=2 flg=0x1100
 n798n    (  0)          lshl (in GPR_0017) (X>=0 )                                                   [       0x11fa0a920] bci=[0,12,180] rc=0 vc=2947 vn=- li=57 udi=14464 nc=2 flg=0x100
 n799n    (  0)            i2l (highWordZero X>=0 )                                                   [       0x11fa0a970] bci=[0,12,180] rc=0 vc=2947 vn=- li=57 udi=- nc=1 flg=0x4100
 n12n     (  0)              ==>iRegLoad (in GPR_0017) (cannotOverflow )
 n803n    (  0)            iconst 1 (Unsigned X!=0 X>=0 )                                             [       0x11fa0aab0] bci=[0,12,180] rc=0 vc=2947 vn=- li=57 udi=- nc=0 flg=0x4104
 n804n    (  0)          lconst -16 (X!=0 X<=0 )                                                      [       0x11fa0ab00] bci=[0,12,180] rc=0 vc=2947 vn=- li=57 udi=- nc=0 flg=0x204
 n805n    (  1)      aladd (in GPR_0113) (X>=0 internalPtr )                                          [       0x11fa0ab50] bci=[0,34,185] rc=1 vc=2947 vn=- li=57 udi=62976 nc=2 flg=0x8100
 n7n      (  0)        ==>newarray (in &GPR_0038) (X!=0 )
 n807n    (  0)        lconst 16 (highWordZero X!=0 X>=0 cannotOverflow )                             [       0x11fa0abf0] bci=[0,34,185] rc=0 vc=2947 vn=- li=57 udi=- nc=0 flg=0x5104
 n1286n   (  1)      iconst 0 (X==0 X>=0 X<=0 )                                                       [       0x11fda41b0] bci=[-1,0,161] rc=1 vc=2947 vn=- li=- udi=- nc=0 flg=0x302
 n816n    (  1)      iconst 0xff00ff00 (X!=0 X<=0 )                                                   [       0x11fa0aec0] bci=[0,10,180] rc=1 vc=2947 vn=- li=57 udi=- nc=0 flg=0x204
 n815n    (  1)      i2l (in GPR_0114) (highWordZero X>=0 )                                           [       0x11fa0ae70] bci=[0,5,179] rc=1 vc=2947 vn=- li=57 udi=63232 nc=1 flg=0x4100
 n5n      (  1)        ==>iRegLoad (in GPR_0018) (cannotOverflow )
 n1287n   (  1)      iconst -1 (X!=0 X<=0 )                                                           [       0x11fda4200] bci=[-1,0,161] rc=1 vc=2947 vn=- li=- udi=- nc=0 flg=0x204
```